File: Servieces/FilterService.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json.Linq;
using Repositories;
using Repositories.Models;

namespace Servieces
{
   public class FilterService: IFilterService
    {
        IFilterRepository filterRepository;
        public FilterService(IFilterRepository filterRepository)
        {
            this.filterRepository = filterRepository;
        }

        public List<JObject> projectName(int companyCode)
        {
            return filterRepository.projectName(companyCode);
        }
        public Project tablesName(int code, string projectName)
        {
            return filterRepository.tablesName(code, projectName);
        }
  public List<string> getColumnName(int projectCode, string tableName)
        {
            return filterRepository.getColumnName(projectCode, tableName);
        }
        //  public Boolean saveFilter(int projectCode,string tableName,List<string> columns)
        //{
        //    return filterRepository.saveFilter(projectCode, tableName, columns);
        //}
        public FilterType getFilterTypes(int projectCode, string tableName, string column)
        {
            return filterRepository.getFilterTypes(projectCode, tableName, column);
        }
       public int saveFilter(dynamic filter)
        {
            return filterRepository.saveFilter(filter);
        }
       public List<string> getNames(dynamic filterToSend)
        {
            return filterRepository.getNames(filterToSend);
        }
        public List<FilterTables> getFilter(int projectCode)
        {
            return filterRepository.getFilter(projectCode);
        }
        public bool deleteFilter(int filterId)
        {
            return filterRepository.deleteFilter(filterId);
        }
    }
}

File: angular-project1/src/app/Services/filter.service.ts
import { Injectable } from '@angular/core';
import { HttpServerService } from '../http-server.service';
import { Filter } from '../models/Filter';
import { FieldFilter } from '../models/FieldFilter';

@Injectable({
  providedIn: 'root'
})
export class FilterService {

  constructor(private http:HttpServerService) { }
  projectName(companycode:string){
    debugger;
return this.http.get('Filter/projectName/' + companycode);
  };
  getTablesName(companycode:string,projectName:string){
    return this.http.get('Filter/tablesName/'+companycode+'/'+projectName );
  };
  getColumnName(projectCode:string,tableName:string){
    debugger;
   return this.http.get('Filter/getColumnName/' +projectCode + '/' + tableName); 
 };
 saveFilter(filter:Filter){
  debugger;
  return this.http.post('Filter/saveFilter' ,filter);
};
saveField(car:{field:FieldFilter,id:number,TableName:string})
{
  return this.http.post('Filter/saveField' ,car);

}
getFilterTypes(projectCode:string,tableName:string,columnName:string){
  debugger;
  return this.http.get('Filter/getFilterTypes/' +projectCode + '/' + tableName+ '/' +columnName);
};
getNames(filterToSend:Filter){
  return this.http.post('Filter/filterToSend' ,filterToSend);
}
getFilter(projectCode:number){
  return this.http.get('Filter/getFilter/' +projectCode);

}
deleteFilter(filterId:number)
{
  return this.http.delete('Filter/deleteFilter/'+filterId);
}
}
